File: Optional4/opt3.cpp
#include <stdio.h>
#include <string.h>
double exp(double num, int expo);
double resultIntNum(char num[], int length);
int checkPointPos(char str[]);
double resultFloatNum(char num[], int begin, int length);
int main() {
	char a[100];
	scanf("%s", a);
	if (!checkPointPos(a))
		printf("The number is %g\n", resultIntNum(a, strlen(a)));
	else 
		printf("The number is %g\n", resultIntNum(a, checkPointPos(a)) + resultFloatNum(a, checkPointPos(a), strlen(a)));
	return 0;
}
double resultIntNum(char num[], int length) {
	double result = 0;
	if (num[0] <= '9' && num[0] >= '0') {
		for (int i = 0; i < length; i++)
			result += ((int)num[length - i - 1] - 48) * exp(10, i);
	}
	else {
		for (int i = 0; i < length - 1; i++)
			result += ((int)num[length - i - 1] - 48) * exp(10, i);
		if (num[0] == '+')
			return result;
		else if (num[0] == '-')
			return -result;
		else
			return 0;
	}
	return result;
}
double exp(double num, int expo) {
	double result = 1;
	for (int i = 0; i < expo; i++)
		result *= num;
	return result;
}
int checkPointPos(char str[]) {
	for (int i = 0; i < strlen(str); i++) {
		if (str[i] == '.')
			return i;
	}
	return 0;
}
double resultFloatNum(char num[], int begin, int length) {
	double result = 0;
	for (int i = begin + 1; i < length; i++) 
		result += ((int)num[i] - 48) * exp(0.1, i - begin);
	if (num[0] == '-')
		return -result;
	return result;
}

File: Optional4/opt3.java
import java.util.Scanner;
import java.text.DecimalFormat;
public class opt3{
    public static void main(String[] args){
        Scanner scan = new Scanner(System.in);
        String number = scan.next();
        if (checkPointPos(number) == 0)
            System.out.printf("The number is %f\n", resultNumber(number, number.length(), checkFirst(number)));
        else
            System.out.printf("The number is %s\n", new DecimalFormat().format(resultNumber(number, checkPointPos(number), checkFirst(number)) + resultFloatNum(number, checkPointPos(number), number.length())));
    }
    private static boolean checkFirst(String str){
        return str.charAt(0) <= '9' && str.charAt(0) >= '0';
    }
    private static double resultNumber(String number, int length, boolean judge){
        double result = 0;
        if (judge){
            for (int i = 0; i < length; i++)
                result += ((int) number.charAt(length-1-i)-48) * Math.pow(10, i);
            return result;
        }
        else{
            for (int i = 0; i < length - 1; i++)
                result += ((int) number.charAt(length-1-i)-48) * Math.pow(10, i);
            if (number.charAt(0) == '+')
                return result;
            else if (number.charAt(0) == '-')
                return -result;
        }
        return 0;
    }
    private static int checkPointPos(String str){
        for (int i = 0; i < str.length(); i++) {
            if (str.charAt(i) == '.')
                return i;
        }
        return 0;
    }
    private static double resultFloatNum(String str, int begin, int length){
        double result = 0;
        for (int i = begin + 1; i < length; i++)
            result += ((int)str.charAt(i) - 48) * Math.pow(0.1, i - begin);
        if (str.charAt(0) == '-')
            return -result;
        return result;
    }
}